R/FilteredData-utils.R

```diff
@@ -9,7 +9,7 @@
 #' library(shiny)
 #' datasets <- teal.slice::init_filtered_data(list(iris = iris, mtcars = mtcars))
 #' @export
-init_filtered_data <- function(x, join_keys = teal.data::join_keys(), code, check) { # nolint
+init_filtered_data <- function(x = list(), join_keys = teal.data::join_keys(), code, check) { # nolint
```

Comment:
- I don't think `x` should have a default value. Usually a default value is a signal that a given argument is not obligatory which is not the case.
- If we are talking about this line - there are `code` and `check` argumnets that are both deprecated and without a default value. I think we can safely add default values. Note that `missing()` used below only checks if a value was _explicitely_ specified in the function call.
``` r
foo <- \(x = NULL) {`if`(missing(x), 0, 1)}
foo()
#> [1] 0
foo(10)
#> [1] 1
```

<sup>Created on 2023-12-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>